### PR TITLE
SI-874 JSR-223 compliance for the interpreter

### PR DESCRIPTION
**Pull request by rjolly**
Usage:

scala/bin/jarlister -o scala-library-listed.jar scala/lib/scala-library.jar

jrunscript -classpath scala/lib/scala-compiler.jar:scala/lib/scala-reflect.jar:scala-library-listed.jar -l scala

### CODE REVIEW

**Comment by JamesIry** [comment]
review @paulp

**Comment by JamesIry** [comment]
PLS SYNCH

**Comment by JamesIry** [comment]
PLS REBUILD ALL

**Comment by JamesIry** [comment]
ping @adriaanm The flag under issues shows this has been tested but it's not updated here

ping @paulp for review

**Comment by adriaanm** [comment]
BUILDLOG?

File: src/compiler/scala/tools/nsc/JarLister.scala [hunk 1]
```diff
@@ -0,0 +1,80 @@
+package scala.tools.nsc
```
Comment (adriaanm) [inline src/compiler/scala/tools/nsc/JarLister.scala:6]: this needs a copyright/author header

File: src/compiler/scala/tools/nsc/JarLister.scala [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+import java.io.{File, FileOutputStream, IOException}
+import java.util.jar.{JarFile, JarEntry, JarOutputStream, Attributes}
+import scala.collection.convert.WrapAsScala
+import WrapAsScala.{enumerationAsScalaIterator, mapAsScalaMap}
+
+class JarLister {
```
Comment (adriaanm) [inline src/compiler/scala/tools/nsc/JarLister.scala:17]: a quick line or two of comment goodness, mayhaps?

**Comment by adriaanm** [comment]
This is great stuff, thanks @rjolly! (I apologize for not having gotten to this faster.)

I'd like to see it split up in several commits, so that if something breaks (one never knows) we can revert more selectively. 

How about the following breakdown:
- reflect.io improvements
- jarlister (but not used in build yet)
- use jarlister in build
- actual JSR-223 implementation

Also, review by @jsuereth for build/osgi-related changes.

**Comment by JamesIry** [comment]
subsumed by https://github.com/scala/scala/pull/2206

**Pull request closed without merging**